ui/admin/app/hooks/knowledge/useKnowledgeSourceFiles.ts

```diff
@@ -15,6 +16,14 @@ export function useKnowledgeSourceFiles(
 ) {
     const [blockPollingFiles, setBlockPollingFiles] = useState(true);
 
+    const startPolling = () => {
+        if (blockPollingFiles) setBlockPollingFiles(false);
+    };
+
+    if (knowledgeSource.state === KnowledgeSourceStatus.Syncing) {
```

Comment:
you should also check for pending state. Something if sync happen to fast it will never fetch state from syncing given we are pulling every 5 seconds

Reply:
fixed! thanks!